### ISSUE TITLE
Audit test pipelines

### ISSUE THREAD

**Issue by cachedout** [elastic/e2e-testing#3053]
This is a master tracking issue issue for auditing which E2E test pipelines need to remain enabled.

## Beats CI pipelines

|Pipeline|Main Health|Triggers|Stakeholders|Issue(s)|Removal planned|
|--------|-------|--------|-------------|-----|:----------:|
|[Docker images](https://beats-ci.elastic.co/job/e2e-tests/job/build-docker-images/) | ⭕ Stale|None|Robots [@cachedout and @kuisathaverat ]||❌|
|[Fleet E2E](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-fleet-daily-mbp/)|🔴 Broken|[Daily build](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-schedule-daily/)|Fleet [@joshdover]|https://github.com/elastic/elastic-agent/issues/1174|
|[Observability Helm Charts](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-helm-daily-mbp/)|🟢 Healthy|[Daily build](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-schedule-daily/)|Robots [@cachedout and @kuisathaverat ]|Issue located in private repo|✅|
|[K8S Autodiscover](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-k8s-autodiscovery-daily-mbp/)|🟡 Flakey|[Daily build](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-schedule-daily/)|Cloud Native Monitoring [@gizas] |
|[Observability MacOS](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-macos-daily-mbp/)|🔴 Broken|[Daily build](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-schedule-daily/)|Elastic Agent [@cmacknz and @jlind23 ]|https://github.com/elastic/ci/issues/705|
|[Fleet Server](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-fleet-server/)|⭕ Stale|None|Fleet [@joshdover ]|https://github.com/elastic/fleet-server/issues/1927|
|[Fleet UI](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-kibana-fleet/)|⭕ Stale|None|Fleet and Integrations [@kpollich ]||

## Fleet CI pipelines
|Pipeline|Main Health|Triggers|Stakeholders|Issue|
|--------|-------|--------|-------------|-----|
|[Pipeline helper](https://fleet-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/)|🔴 Broken|Push to main; PR labeled|Elastic Agent[@cmacknz and @jlind23 ]|https://github.com/elastic/elastic-agent/issues/1174|

⚠️ If you are listed as a stakeholder, we would like to know the following:

1. Should the pipeline be removed from the CI or should it remain?
1.1 If the pipeline remains and is broken, what is the link to an issue tracking a fix?
1.2 If the pipeline should remain, how is it monitored by the team to ensure that build artifacts are not produced when the tests fail?

### Proposed pipeline criteria
I am proposing that we remove all pipelines which do not meet **any** of the following criteria:

1. Necessary for the ongoing health of the E2E test suite itself
2. Used by a product team as a quality gateway. Concretely, this means that a failing test blocks a PR from being merged or a build artifact from being produced.
3. Exist to ensure the quality of a supported product. 

### Timeline
1. All existing E2E pipelines have stakeholders assigned no later than: **October 1, 2022**
2. All stakeholder agree upon proposed pipeline criteria no later than: **October 20, 2022**
3. Non-confirming pipelines will be removed from Jenkins and code will be removed from the E2E test suite beginning on: **Nov 1st, 2022**

## Related efforts

There is a separate effort to try and reduce the scope of E2E testing back to a point where stability can be maintained, but it is limited to tests for the Agent. That effort can be found here: https://github.com/elastic/elastic-agent/issues/1174

**Comment by v1v** [comment]
For the Macos Daily -> it was originally implemented in https://github.com/elastic/e2e-testing/pull/2626, and using the Orka ephemeral workers, and superseded  https://github.com/elastic/e2e-testing/pull/233

the error is something the @elastic/ci-systems might need to help with:

```
[2022-09-28T04:58:54.763Z] + .ci/scripts/deployment.sh create
[2022-09-28T04:58:54.887Z] Cloning into '.obs'...
[2022-09-28T04:58:55.095Z] Host key verification failed.
[2022-09-28T04:58:55.095Z] fatal: Could not read from remote repository.
[2022-09-28T04:58:55.095Z] 
[2022-09-28T04:58:55.095Z] Please make sure you have the correct access rights
[2022-09-28T04:58:55.095Z] and the repository exists.
```

IIUC, the recent upgrade in the CI controllers added a host key verification by default, we reported this in the past and it was partially fixed since we dont' see the below error but a new one:

<img width="1607" alt="image" src="https://user-images.githubusercontent.com/2871786/192753634-8c74aab2-d314-4318-b2ca-059bc212b584.png">

but the error now happens in a subsequent stage to clone a private repository -- see the above console log

It [worked](https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-macos-daily-mbp/detail/main/31/pipeline) in the past 

<img width="1239" alt="image" src="https://user-images.githubusercontent.com/2871786/192753980-3aab087d-9fb7-4414-90a6-625b04616cbf.png">

**Comment by kuisathaverat** [comment]
[Docker images](https://beats-ci.elastic.co/job/e2e-tests/job/build-docker-images/) generated the Systemd Docker images used in the e2e tests, probably we are the stakeholders.

**Comment by cachedout** [comment]
@v1v Thanks, that helps. I'm also trying to figure out what it actually _does_ so that I can figure out how the stakeholders should be. I'm code-diving right now a bit to try and get a sense of that.

**Comment by kuisathaverat** [comment]
[Observability Helm Charts](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-helm-daily-mbp/) can be removed

**Comment by cachedout** [comment]
@kuisathaverat Thanks! Regarding the Docker images -- that pipeline hasn't been executed for over a year. Does it still need to exist?

**Comment by kuisathaverat** [comment]
>Does it still need to exist?

It is the only way to generate those images, when they change should be executed. These images are for making a test on installation on a systems environment. The main change that can have is bump the systems version or the Linux version.

**Comment by cachedout** [comment]
@cmacknz and @jlind23 Are you tracking any issues for the flakiness in the [K8s Autodiscover pipeline](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-k8s-autodiscovery-daily-mbp/)?

**Comment by v1v** [comment]
> @v1v Thanks, that helps. I'm also trying to figure out what it actually does so that I can figure out how the stakeholders should be. I'm code-diving right now a bit to try and get a sense of that.

There was an original request to test on MacOS, for such, it was initially attempted with the AWS MacOS, but it was declined for vary reasons:
1) Cost, IIRC, machines will be created and pay for 24 hours minimal, see https://github.com/elastic/e2e-testing/pull/2336#issuecomment-1111883732
2) Implementation, the Ansible ec2 integration didn't work well , see https://github.com/elastic/e2e-testing/pull/2336#issuecomment-1118315327
3) Ephemeral Orkas were available. see https://github.com/elastic/e2e-testing/pull/2336#issuecomment-1147931612


I guess the stakeholder might be @jlind23 as he was the original requester for the MacOS in AWS

**Comment by jlind23** [comment]
@cachedout this is the [issue](https://github.com/elastic/e2e-testing/issues/3053) we will use for the first half of 8.6. @AndersonQ is already assigned to this and will closely work with you in order to get back to a better place.

**Comment by cachedout** [comment]
@jlind23 That link seems wrong? :)

**Comment by jlind23** [comment]
Sorry, this one - https://github.com/elastic/elastic-agent/issues/1174

**Comment by cmacknz** [comment]
> @cmacknz and @jlind23 Are you tracking any issues for the flakiness in the [K8s Autodiscover pipeline](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-k8s-autodiscovery-daily-mbp/)?

No, it may make sense to follow up with the Observability Cloudnative monitoring team to see if they have interest in fixing these tests faster than the agent team can get to them. They have done the majority of the recent work for autodiscovery features in agent.

**Comment by cachedout** [comment]
> No, it may make sense to follow up with the Observability Cloudnative monitoring team to see if they have interest in fixing these tests faster than the agent team can get to them.

Looping in @gizas . We are trying to stabilize the E2E test suite. Are you aware of the flakiness in the k8s autodiscover tests, and if so, is anybody on your time investigating them?

**Comment by gizas** [comment]
Sorry for delayed answer, @cachedout , @cmacknz just checking [K8s Autodiscover pipeline](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-k8s-autodiscovery-daily-mbp/). Can you point me to a fail instance to have a look?

Indeed in the past we had provided some fixes

**Comment by joshdover** [comment]
> [Observability Helm Charts](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-helm-daily-mbp/) can be removed

Any reason this hasn't been done yet? Seeing it fail on a few PR runs recently and couldn't find the issue to track removing these.

**Comment by cachedout** [comment]
> Any reason this hasn't been done yet? 

Hi @joshdover . The issue is this one: https://github.com/elastic/observability-robots/issues/1325

We were considering this blocked until they sorted out the public communication regarding chart deprecation, but TBH it's probably not a big deal if we just pull it out now if it's failing in PRs. LMK what you think.

**Comment by joshdover** [comment]
Makes sense. I've only seen it fail once recently, but will flag it if it's
more of a problem.
